palette/src/convert.rs

```diff
@@ -499,6 +502,173 @@ where
     }
 }
 
+///The error type for a color conversion that converted a color into a color with invalid values.
+#[derive(Debug)]
+pub struct OutOfBounds<T> {
+    color: T,
+}
+
+impl<T> OutOfBounds<T> {
+    ///Create a new error wrapping a color
+    fn new(color: T) -> Self {
+        OutOfBounds { color }
+    }
+
+    ///Consume this error and return the wrapped color
+    pub fn color(self) -> T {
+        self.color
+    }
+}
+
+impl<T: Debug> error::Error for OutOfBounds<T> {
+    fn description(&self) -> &str {
+        "Color conversion is out of bounds"
+    }
+}
+
+impl<T> Display for OutOfBounds<T> {
+    fn fmt(&self, fmt: &mut Formatter) -> fmt::Result {
+        write!(fmt, "Color conversion is out of bounds")
```

Comment:
You could reuse the description here: `self.description().fmt(fmt)`.

Reply:
I didn't implement description at all at first because it is `soft deprecated` as its docs say, hence I didnt use it there either, but I might as well. No need to duplicate the literal there.